tests/test_regularizer.py

```diff
@@ -414,14 +555,17 @@ def test_solver_match_sklearn_gamma(self, gammaGLM_model_instantiation):
         # set precision to float64 for accurate matching of the results
         model.data_type = jnp.float64
         model.observation_model.inverse_link_function = jnp.exp
-        regularizer = self.cls("GradientDescent", {"tol": 10**-12})
-        regularizer.regularizer_strength = 0.1
-        runner_bfgs = regularizer.instantiate_solver(model._predict_and_compute_loss)[2]
+        model.regularizer = self.cls()
+        model.solver_kwargs = {"tol": 10**-12}
+        model.regularizer_strength = 0.1
+        runner_bfgs = model.instantiate_solver()[2]
         weights_bfgs, intercepts_bfgs = runner_bfgs(
             (true_params[0] * 0.0, true_params[1]), X, y
         )[0]
         model_skl = GammaRegressor(
-            fit_intercept=True, tol=10**-12, alpha=regularizer.regularizer_strength
+            fit_intercept=True,
+            tol=10**-12,
+            alpha=model.regularizer_strength,
```

Comment:
```suggestion
            alpha=float(model.regularizer_strength),
```

This is needed if we go for the tree_map in the setter of reg strength